Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -223,29 +222,73 @@ class InAppMessageViewTest: UnitTest {
         queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
 
         let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
-        await onDoneRenderingInAppMessage(givenInlineMessage)
-        XCTAssertTrue(isDisplayingInAppMessage(inlineView))
+        await onDoneRenderingInAppMessage(givenInlineMessage, insideOfInlineView: inlineView)
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        await onCloseActionButtonPressed(onInlineView: inlineView)
+
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+        XCTAssertNil(getInAppMessage(forView: inlineView))
+    }
+
+    @MainActor
+    func test_onCloseAction_givenMultipleMessagesInQueue_expectDisplayNextMessageInQueueAfterClose() async {
+        let givenElementId = String.random
+        let givenMessages = [Message(elementId: givenElementId), Message(elementId: givenElementId)]
+        queueMock.getInlineMessagesReturnValue = givenMessages
+
+        let view = InAppMessageView(elementId: givenElementId)
+
+        await onDoneRenderingInAppMessage(givenMessages[0], insideOfInlineView: view)
+        XCTAssertTrue(isInlineViewVisible(view))
+        XCTAssertEqual(getInAppMessage(forView: view), givenMessages[0])
 
-        await onCloseActionButtonPressed()
+        await onCloseActionButtonPressed(onInlineView: view)
+        await onDoneRenderingInAppMessage(givenMessages[1], insideOfInlineView: view)
+        XCTAssertTrue(isInlineViewVisible(view))
+        XCTAssertEqual(getInAppMessage(forView: view), givenMessages[1])
 
-        XCTAssertFalse(isDisplayingInAppMessage(inlineView))
+        await onCloseActionButtonPressed(onInlineView: view)
+        XCTAssertFalse(isInlineViewVisible(view))
+        XCTAssertNil(getInAppMessage(forView: view))
     }
 
-    // Once an in-app message has been closed it will not be replaced with another message.
-    // We plan to change this behavior in the future. Test function can be modified to match the new behavior at that time.
     @MainActor
-    func test_onCloseAction_givenMessageClosed_givenNewMessageFetched_expectIgnoreMessage() async {
+    func test_onCloseAction_givenMessageClosed_givenNewMessageFetched_expectDisplayNewMessage() async {
```

Comment:
Test function modified to match new expected behavior. 

If we close a message and then a new message is fetched in the future, that new fetched message will be shown.